public/scripts/report.js

```diff
@@ -1,24 +1,80 @@
 (function(){
 
-  // GET LOCATION
-  var locationButton = document.getElementById('getLocationButton');
-
-  function getLocation() {
-    if (navigator.geolocation) {
-      navigator.geolocation.getCurrentPosition(showPosition);
-    } else {
-      // non-supprt error handling
-      console.log('Geolocation is not supported by this browser.');
+  // GET USER'S LOCATION
+  function getLocation(){
+
+    var locationButton = document.getElementById('getLocationButton');
+
+    function getLocation() {
+      if (navigator.geolocation) {
+        navigator.geolocation.getCurrentPosition(storePosition);
+      } else {
+        // non-supprt error handling
+        console.log('Geolocation is not supported by this browser.');
+      }
+    };
+    // store user's location in button attribute
+    function storePosition(position) {
+      var getLocationButton = document.getElementById('getLocationButton');
+
+      var coords = {
+        'latitude': position.coords.latitude,
+        'longitude': position.coords.longitude
+      };
+
+      getLocationButton.setAttribute("value", JSON.stringify(coords));
+
+      getLocationButton.innerHTML = 'Using current location';
```

Comment:
Maybe also set innerHTML to 'getting location' or something while this process is happening